### PR TITLE
added Usage Type and Exemption tags in cookiecutter for tier4

### PR DESCRIPTION
**Pull request by sachin-panayil**
## Problem

Usage Type and Exemption tags were not included inside the cookiecutter.

## Solution

Add the following tags within the cookiecutter.

## Result

code.json will not ask for the following tags when it is run.

Some important notes regarding the summary line:

* only added in tier 4
* rest of tiers will be updated once tier 4 is approved

## Test Plan

run cookiecutter commands to check that the new tags are appearing properly.

### CODE REVIEW

File: tier4/{{cookiecutter.project_slug}}/.github/codejson/{{cookiecutter.project_name}}/code.json
Comment (natalialuzuriaga) [inline tier4/{{cookiecutter.project_slug}}/.github/codejson/{{cookiecutter.project_name}}/code.json:37]: As per the code.json schema, could `exemption` be changed to `exemptionText`? Also update accordingly in `cookiecutter.json`

Comment (natalialuzuriaga) [reply tier4/{{cookiecutter.project_slug}}/.github/codejson/{{cookiecutter.project_name}}/code.json:37]: https://github.com/GSA/code-gov-data/blob/master/schemas/schema-2.0.0.json#L97

File: tier4/{{cookiecutter.project_slug}}/.github/cookiecutter.json
Comment (natalialuzuriaga) [inline tier4/{{cookiecutter.project_slug}}/.github/cookiecutter.json:20]: Could we have the `usage_type` values match the code.json values? We would like to directly follow the federal-wide schema

https://github.com/GSA/code-gov-data/blob/master/schemas/schema-2.0.0.json#L86

File: tier4/{{cookiecutter.project_slug}}/.github/cookiecutter.json
Comment (natalialuzuriaga) [inline tier4/{{cookiecutter.project_slug}}/.github/cookiecutter.json:21]: As per the schema, `exemptionText` is required if the `usageType` selected is an exemption value (exemptByLaw, exemptyByCIO, etc.) 

Is there a way to add this logic in `post_gen_hook.py` or in `cookiecutter.json` where we ask the user to fill out the `exemptionText` only if they have answered with an exemption value for `usageType`? Not sure if cookiecutter can support this conditionals hmmmmm

**Review by natalialuzuriaga — CHANGES_REQUESTED** [review]
Solid start!! Couple of minor changes that have to do with aligning with the code.json schema. Also not sure if it is possible to add some conditional type logic into the cookiecutter prompts so lmk what you find!

File: tier4/{{cookiecutter.project_slug}}/.github/cookiecutter.json
Comment (sachin-panayil) [reply tier4/{{cookiecutter.project_slug}}/.github/cookiecutter.json]: i attempted to do this within the JSON using Jinja2 resources but I couldnt get to a solution. ended up putting the logic within post_gen hook so it asks the to enter an exemption text if they choose an exemption AND the user is finished filling out the cookiecutter

**Comment by sachin-panayil** [comment]
im also merging main into dev with this branch!

**Review by decause-gov — APPROVED** [review]
May want Murt/Nat to give it a look too, but LGTM +1 🚢

**Comment by sachin-panayil** [comment]
gonna hold off to merge until Nat takes a look bc she was spearheading this ticket @decause-gov

**Comment by decause-gov** [comment]
[like]  DeCausemaker, Remy (CMS/DSAC) reacted to your message:
________________________________

File: tier4/{{cookiecutter.project_slug}}/.github/codejson/hooks/post_gen_project.py
Comment (natalialuzuriaga) [inline tier4/{{cookiecutter.project_slug}}/.github/codejson/hooks/post_gen_project.py:41]: Perfect, I like how this is done 🙌

File: tier4/{{cookiecutter.project_slug}}/.github/codejson/{{cookiecutter.project_name}}/code.json
Comment (natalialuzuriaga) [inline tier4/{{cookiecutter.project_slug}}/.github/codejson/{{cookiecutter.project_name}}/code.json:36]: Please add `usageType` and `exemptionText` inside the `permissions` object as shown in the code.json schema here:
https://github.com/GSA/code-gov-data/blob/master/schemas/schema-2.0.0.json
https://github.com/GSA/code-gov/blob/master/docs/metadata_examples.md

File: tier4/{{cookiecutter.project_slug}}/.github/codejson/hooks/post_gen_project.py
Comment (natalialuzuriaga) [inline tier4/{{cookiecutter.project_slug}}/.github/codejson/hooks/post_gen_project.py:42]: Could we add the exemption value the user selected to this print statement here, just so they know what we are referring to

**Review by natalialuzuriaga — CHANGES_REQUESTED** [review]
Tested out the `exemptionText` flow and it works great! A minor change to the flow is needed plus one more code.json change so that our code.json matches the federal-wide schema

**Review by natalialuzuriaga — APPROVED** [review]
Great work, tested and it all LGTM 🙌 🚢

**Pull request merged**
